tools/normalize.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020 Codethink Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <string.h>
+
+#include "../src/msg.h"
+
+#include "msg.h"
+#include "sig.h"
+#include "fifo.h"
+
+// DEFAULT_AVERAGE_WIDTH, the number of milliseconds to average over
+long DEFAULT_AVERAGE_WIDTH = 1000;
+
+struct channel_data {
+	bl_msg_sample_data_t msg; // Used as a buffer and for outputting the processed samples
+	struct fifo samples;      // Store samples until we've finished using them for processing
+	double rolling_average;   // The rolling average as of the most recent sample
```

Comment:
Hmm, I'd need to experiment to know if it would work, but maybe:

```suggestion
	uint64_t rolling_sum;     // The rolling sum as of the most recent sample
```